.coveragerc

```diff
@@ -10,6 +10,9 @@ omit =
     homeassistant/helpers/signal.py
 
     # omit pieces of code that rely on external devices being present
+    homeassistant/components/azure/__init__.py
+    homeassistant/components/*/azure_vm.py
```

Comment:
Please rename the platform module to have the same name as the package.

Reply:
So the idea is that there will be multiple sensors based on azure. This sensor platform is just for azure virtual machine so it is called azure_vm. I will soon have other types of sensors like azure_service_bus, azure_storage_table. They will depend on the azure component.

Reply:
Then they will be other sensor types but within the same platform module.

Reply:
So to achieve something like 

sensor:
  - platform: azure_vm
    name: hassio-test-vm
    resource_group: hassio-test-resource-group
  - platform: azure_servicebus
    name: hassio-test-servicebus
    type: queue

don't they have to be in files named azure_vm.py and azure_servicebus.py ?

Reply:
We will only use one platform but have different monitored conditions (sensor types). See monitored_conditions in the source for examples. At least that's what I would recommend.

Reply:
Do you think that will work well in the long term for something like azure. Monitored conditions seems to be built for something like a sensor giving multiple types of measurements. Like BME280 doing Humidity and Temperature measurements. Azure being a cloud service has vastly disparate service offerings(some not even related to one another). Having them all in one module will quickly complicate stuff.

Reply:
The closest example I can site is under notify component we have aws_lambda.py, aws_sns.py and aws_sqs.py. Even tough they are all AWS each is in its own module.

Reply:
Then open an issue in our architecture repo and explain your intention.

We don't have any two platforms currently that belong to the same mother component.

Reply:
I have created https://github.com/home-assistant/architecture/issues/68.

Reply:
@MartinHjelmare how do we proceed now ?

Reply:
We can't move forward here like this until we've gotten approval in the architecture issue. If we change here to the normal layout we can go ahead though.

Reply:
Don't forget to update this path.